packages/foam-vscode/src/features/create-from-template.ts

```diff
@@ -283,6 +283,17 @@ export async function resolveFoamTemplateVariables(
   );
   const uniqVariables = [...new Set(variables)];
 
+  const missingFoamVariables = [...givenValues.keys()].filter(
+    x => uniqVariables.indexOf(x) < 0
+  );
+  if (missingFoamVariables.length > 0) {
+    missingFoamVariables.forEach(key => {
+      templateText = `${templateText}
+      \${${key}}`;
```

Comment:
do I understand correctly that you are appending all un-referenced variables to the template?
This behavior should only occur for `FOAM_SELECTED_TEXT`, we treat this variable a "special case" in this regard

Reply:
Changed the behaviour. In hindsight it indeed makes more sense to only implicitly add `FOAM_SELECTED_TEXT`.